spec.go

```diff
@@ -311,26 +311,47 @@ func createLibcontainerMount(cwd string, m specs.Mount) *configs.Mount {
 	}
 }
 
-func createCgroupConfig(name string, spec *specs.Spec) (*configs.Cgroup, error) {
+func createCgroupConfig(name string, useSystemdCgroup bool, spec *specs.Spec) (*configs.Cgroup, error) {
 	var (
 		err          error
 		myCgroupPath string
 	)
 
+	c := &configs.Cgroup{
+		Resources: &configs.Resources{},
+	}
+
 	if spec.Linux.CgroupsPath != nil {
 		myCgroupPath = libcontainerUtils.CleanPath(*spec.Linux.CgroupsPath)
+	}
+
+	if useSystemdCgroup {
+		if myCgroupPath == "" {
+			c.Parent = "system.slice"
+			c.ScopePrefix = "runc"
+			c.Name = name
+		} else {
+			// Parse the path from expected "slice:prefix:name"
+			// for e.g. "system.slice:docker:1234"
+			parts := strings.Split(myCgroupPath, ":")
```

Comment:
@anusha-ragunathan Yes, I am well aware of that. The choice to encode the parameters within cgroupsPath was deliberate. The reason is that systemd cgroups API doesn't work with full cgroups paths. See https://github.com/opencontainers/runc/blob/master/libcontainer/cgroups/systemd/apply_systemd.go#L227

It expects the unit name that is formed from the prefix and name and the parent slice name.
Since OCI doesn't support these as first class fields, these are encoded in the cgroupsPath using ":" as a separator as a practical choice. 

Also, I don't think that these should be first class concepts in OCI and runc can add this to support systemd cgroups as it isn't bound to just supporting OCI.